libs/auth/src/drivers/magento/login.service.spec.ts

```diff
@@ -0,0 +1,130 @@
+
+  const registrationFactory: DaffAccountRegistrationFactory = new DaffAccountRegistrationFactory();
+  const authTokenFactory: DaffAuthTokenFactory = new DaffAuthTokenFactory();
+
+  let mockAuth: DaffAuthToken;
+  let mockLoginInfo: DaffLoginInfo;
+  let token: string;
+  let email: string;
+  let password: string;
+  let firstName: string;
+  let lastName: string;
+  let mockRegistration: DaffAccountRegistration<DaffCustomerRegistration>;
+
+  let authGraphQlQueryManagerService: DaffMagentoAuthGraphQlQueryManagerService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        ApolloTestingModule
+      ],
+      providers: [
+        DaffMagentoLoginService,
+        {
+          provide: DaffAuthQueryManager,
+          useExisting: DaffMagentoAuthGraphQlQueryManagerService
+        },
+        {
+          provide: DaffAuthTransformer,
+          useValue: transformerServiceSpy
+        }
+      ]
+    });
+
+    loginService = TestBed.get(DaffMagentoLoginService);
+    controller = TestBed.get(ApolloTestingController);
+    authGraphQlQueryManagerService = TestBed.get(DaffMagentoAuthGraphQlQueryManagerService);
+
+    mockRegistration = registrationFactory.create();
+    mockAuth = authTokenFactory.create();
+
+    token = mockAuth.token;
+    firstName = mockRegistration.customer.firstName;
+    lastName = mockRegistration.customer.lastName;
+    email = mockRegistration.customer.email;
+    password = mockRegistration.password;
+    mockLoginInfo = {email, password};
+  });
+
+  it('should be created', () => {
+    expect(loginService).toBeTruthy();
+  });
+
+  describe('login | getting an access token', () => {
+    let response;
+
+    afterEach(() => {
+      controller.verify();
+    });
+
+    beforeEach(() => {
+      response = {
+        generateCustomerToken: {
+          token
+        }
+      };
+
+      transformerServiceSpy.transform.withArgs(response).and.returnValue(mockAuth);
+    });
+
+    // TODO: determine if mocked tests are appropriate
+    // it('should call the transformer', () => {
+    //   loginService.login(mockLoginInfo).subscribe((auth) => {
+    //     expect(auth).toEqual(mockAuth);
+    //     // need to do the expectation in here because the transformer is invoked inside a callback
+    //     expect(transformerServiceSpy.transform.calls.any()).toBeTruthy();
+    //   });
+
+    //   const op = controller.expectOne(authGraphQlQueryManagerService.generateATokenMutation(email, password).mutation);
+
+    //   op.flush({
+    //     data: response
+    //   });
+    // });
+
+    it('should return the correct observable', done => {
+      loginService.login(mockLoginInfo).subscribe((auth) => {
+        expect(auth).toEqual(mockAuth);
+        done();
+      });
+
+      const op = controller.expectOne(authGraphQlQueryManagerService.generateATokenMutation({email, password}).mutation);
```

Comment:
probably use the mockLoginInfo here instead, since you already have it defined `generateATokenMutation(mockLoginInfo)`

Reply:
ah good catch